lib/utils/getCacheFile.js

```diff
@@ -0,0 +1,74 @@
+"use strict"
+
+const hash = require("./hash")
+const path = require("path")
+const fs = require("fs")
+/**
+ * return the cacheFile to be used by stylelint, based on whether the provided parameter is
+ * a directory or looks like a directory (ends in `path.sep`), in which case the file
+ * name will be the `cacheFile/.cache_hashOfCWD`
+ *
+ * if cacheFile points to a file or looks like a file then in will just use that file
```

Comment:
I'd like to see capitalization and punctuation in all of these comments. If English is not your first language, feel free to ask for help and someone can pitch in.